android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/item/WebExtensionBrowserMenuItem.kt

```diff
@@ -51,40 +51,52 @@ class WebExtensionBrowserMenuItem(
 
     @Suppress("TooGenericExceptionCaught")
     override fun bind(menu: BrowserMenu, view: View) {
+        val container = view.findViewById<View>(R.id.container)
+        updateItem(view)
+        container.setOnClickListener {
+            listener.invoke()
+            menu.dismiss()
+        }
+    }
+
+    override fun invalidate(view: View) {
+        val labelView = view.findViewById<TextView>(R.id.action_label)
+        val badgeView = view.findViewById<TextView>(R.id.badge_text)
+        val imageView = view.findViewById<ImageView>(R.id.action_image)
+
+        updateItem(view)
```

Comment:
@willdurand I think I found the source of the [bug 1806838](https://bugzilla.mozilla.org/show_bug.cgi?id=18068380, in invalidate we where updating the `label` and `badge` but no the icon, that was causing the previous icon was kept when invalidate is called.

Reply:
Every time we get a new [browser or page action ](https://github.com/mozilla-mobile/firefox-android/blob/35ce01367157440f9e9daa4ed48a8022af80c8f2/android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt#L42-L43), we update the 
the [browserMenuItem.action](https://github.com/mozilla-mobile/firefox-android/blob/35ce01367157440f9e9daa4ed48a8022af80c8f2/android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt#L135) which contains all the data related to the browser or page action, and finally we call [invalidate](https://github.com/mozilla-mobile/firefox-android/blob/35ce01367157440f9e9daa4ed48a8022af80c8f2/android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt#L43) to update the data, I think in some cases, we were reusing the UI items and as the icon was not updated on invalidate, we were keeping the previous icon.

Reply:
I was able to reproduce the issue by [following this report's steps](https://bugzilla.mozilla.org/show_bug.cgi?id=1806838#c27), I installed all the addons the reported had and opened the add-ons menu, put Fenix in background wait for about a minute and then opened it again, and I was able to see icon duplication.


<img width="401" alt="image" src="https://github.com/mozilla-mobile/firefox-android/assets/773158/a9d87ec4-4264-4fc1-a5e3-071f7de92a1f">